src/dft/backends/mklcpu/backward.cpp

```diff
@@ -30,31 +30,143 @@
+#include "mkl_dfti.h"
+
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace mklcpu {
+namespace detail {
 
 // BUFFER version
+// backward a MKLCPU DFT call to the backend, checking that the commit impl is valid.
+template <dft::precision prec, dft::domain dom>
+inline void check_bwd_commit(dft::descriptor<prec, dom> &desc) {
+    auto commit_handle = dft::detail::get_commit(desc);
+    if (commit_handle == nullptr || commit_handle->get_backend() != backend::mklcpu) {
+        throw mkl::invalid_argument("DFT", "computer_backward",
+                                    "DFT descriptor has not been commited for MKLCPU");
+    }
+
+    auto mklcpu_desc = reinterpret_cast<detail::mklcpu_desc_t *>(commit_handle->get_handle());
+    MKL_LONG commit_status{ DFTI_UNCOMMITTED };
+    DftiGetValue(mklcpu_desc[1], DFTI_COMMIT_STATUS, &commit_status);
+    if (commit_status != DFTI_COMMITTED) {
+        throw mkl::invalid_argument("DFT", "compute_backward",
+                                    "MKLCPU DFT descriptor was not successfully committed.");
+    }
+}
+
+// Throw an mkl::invalid_argument if the runtime param in the descriptor does not match
+// the expected value.
+template <dft::detail::config_param Param, dft::detail::config_value Expected, typename DescT>
+inline auto expect_config(DescT &desc, const char *message) {
+    dft::detail::config_value actual{ 0 };
+    desc.get_value(Param, &actual);
+    if (actual != Expected) {
+        throw mkl::invalid_argument("DFT", "compute_backward", message);
+    }
+}
+// convert the base commit class to derived cpu commit class
+template <dft::precision prec, dft::domain dom>
+auto get_buffer(commit_t<prec, dom> *commit_handle) {
+    commit_derived_t<prec, dom> *derived_commit =
+        static_cast<commit_derived_t<prec, dom> *>(commit_handle);
+    return derived_commit->get_handle_buffer();
+}
+} // namespace detail
 
 //In-place transform
 template <typename descriptor_type, typename data_type>
 ONEMKL_EXPORT void compute_backward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout) {
-    throw mkl::unimplemented("DFT", "compute_backward", "Not implemented for MKLCPU");
+    detail::expect_config<dft::detail::config_param::PLACEMENT, dft::detail::config_value::INPLACE>(
+        desc, "Unexpected value for placement");
+    auto commit_handle = dft::detail::get_commit(desc);
+    detail::check_bwd_commit(desc);
+    sycl::queue &cpu_queue{ commit_handle->get_queue() };
+
+    auto mklcpu_desc_buffer{ detail::get_buffer(commit_handle) };
+
+    cpu_queue.submit([&](sycl::handler &cgh) {
+        auto desc_acc = mklcpu_desc_buffer.template get_access<sycl::access::mode::read>(cgh);
+        auto inout_acc = inout.template get_access<sycl::access::mode::read_write>(cgh);
+        detail::host_task<class host_kernel_back_inplace>(cgh, [=]() {
+            DFT_ERROR status;
+            status = DftiComputeBackward(desc_acc[0][detail::DIR::bwd], inout_acc.get_pointer());
```

Comment:
```suggestion
            DFT_ERROR status = DftiComputeBackward(desc_acc[0][detail::DIR::bwd], inout_acc.get_pointer());
```

Reply:
and all the other instances

Reply:
done in 86c1ec999fca920b2212e90b4dad63e56f367b10